frontend/javascripts/oxalis/view/right-border-tabs/nuclei_inferral_modal.js

```diff
@@ -0,0 +1,120 @@
+// @flow
+import React, { useEffect, useState } from "react";
+import { type APIDataset } from "types/api_flow_types";
+import { Modal, Select, Button } from "antd";
+import { startNucleiInferralJob } from "admin/admin_rest_api";
+import { getColorLayers } from "oxalis/model/accessors/dataset_accessor";
+import Toast from "libs/toast";
+import { Unicode } from "oxalis/constants";
+
+const { ThinSpace } = Unicode;
+
+type Props = {
+  dataset: APIDataset,
+  handleClose: () => void,
+};
+
+export default function NucleiInferralModal(props: Props) {
+  const { dataset, handleClose } = props;
+  const [selectedColorLayerName, setSelectedColorLayerName] = useState(null);
+  const colorLayers = getColorLayers(dataset);
```

Comment:
Just a suggestion: Everything in this component is solely based on the names of the color layers. Thus it might be nice to simply map here to the names and then you could get rid of the multiple `.name`-parts below in the file.

I am also fine with keeping it as it is :)